### ISSUE TITLE
[Request] Add upgrade schema for legacy block ids and data values

### ISSUE THREAD

**Issue by KCodeYT** [pmmp/BedrockBlockUpgradeSchema#2]
Hey please add an legacy block state upgrader which upgrades blocks with the following data to the latest block state format.

The block state updater of CloudburstMC ([legacy_block_data_map.json](https://github.com/CloudburstMC/BlockStateUpdater/blob/master/src/main/resources/legacy_block_data_map.json)) already has this feature implemented, but im not sure if it covers all data values and its not licensed under the public domain.

So if I give the nbt:
```json
{
  "name": "minecraft:stone",
  "val": 1
}
```
it should return
```json
{
  "name": "minecraft:stone",
  "states": {
    "stone_type": "granite"
  },
  "version": 17959425
}
```

**Comment by dktapps** [comment]
There already is one. `1.12.0_to_1.18.10_blockstate_map.bin` is what you're looking for.

**Comment by KCodeYT** [comment]
Oh sorry, then for opening an issue sry, but how can I read the format is there an documentation for it?

**Comment by dktapps** [comment]
It's the same as `r12_to_current_block_map.bin` from BedrockData, which is described [here](https://github.com/pmmp/BedrockData#r12_to_current_block_mapbin).

**Comment by KCodeYT** [comment]
Thanks! c:

**Comment by dktapps** [comment]
Please also note that you should process the result of the mapping according to the incremental schemas, since the mapping file won't be updated beyond 1.18.10 (I may replace it with a 1.12 -> 1.13 mapping for maximum usefulness).

**Comment by KCodeYT** [comment]
Yea I also was wondering why its not 1.12 -> 1.13

**Comment by dktapps** [comment]
A couple of reasons. First is because I couldn't be bothered to dig up a 1.13 mapping, second is because some of the mappings changed after 1.13 (bugs were fixed etc), so a 1.12 -> 1.13 mapping as of 1.13 might not be the most desirable thing. More research is needed to go back in time without causing problems.

**Comment by dktapps** [comment]
It just occurred to me that a mapping file for 1.12  -> 1.13 could be easily generated by taking a 1.12 palette from the archive and applying the 1.13 upgrade schema to it. Such an obvious solution, I don't know why I didn't think of it sooner ...
